sdk/python/lib/pulumi/output.py

```diff
@@ -437,22 +437,24 @@ def secret(val: Input[T]) -> "Output[T]":
     # https://mypy.readthedocs.io/en/stable/more_types.html#type-checking-the-variants:~:text=considered%20unsafely%20overlapping
     @overload
     @staticmethod
-    def all(*args: "Output[T_co]") -> "Output[List[T_co]]": ...  # type: ignore
+    def all(*args: "Output[Any]") -> "Output[List[Any]]": ...  # type: ignore
```

Comment:
Previous types stated that all arguments have the same type, however you can pass mixed arguments. Because this returns a list type, and not a tuple, we can't provide overloads for specific types like we do in nodejs https://github.com/pulumi/pulumi/blob/dbb887b5658097dd588931a6d4c7d9cc6d021312/sdk/nodejs/output.ts#L612

[pyright playground example using a tuple return]

https://github.com/pulumi/pulumi/issues/16194

Reply:
Yeh this is kinda sad but probably the most practical we can do for now